### PR TITLE
misc: receive afterSpec durations from app capture code, and report them to the Cloud API

### PR DESCRIPTION
**Pull request by cacieprins**
- Closes 

### Additional details
Because of the bifurcation of Test Replay capture/upload logic, the durations that are recorded in the afterSpec process will be returned from the protocol capture codebase' afterSpec method. They're collected and reported to the `updateInstanceArtifacts` Cloud API endpoint.

### Steps to test
Verify that, once the related services PR is merged and released, these durations are reported to the update instance artifacts endpoint when recording a spec with Test Replay enabled. When running against staging, these durations should be queryable in Honeycomb.

### PR Tasks

- [X] Have tests been added/updated?
- [NA] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [NA] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: packages/server/lib/cloud/api/index.ts [hunk 1]
```diff
@@ -289,6 +290,9 @@ export type ProtocolMetadata = ArtifactMetadata & {
     size: number
     offset: number
   }
+  afterSpecDurations?: AfterSpecDurations & {
```
Comment (cacieprins) [inline packages/server/lib/cloud/api/index.ts:293]: `AfterSpecDurations` is the shape that the protocol capture codebase will return; we're augmenting that with the total duration of `afterSpec`, so the shape is modified here.

File: packages/server/lib/cloud/api/index.ts [hunk 2]
```diff
@@ -508,6 +512,8 @@ export default {
   },
 
   updateInstanceArtifacts (options: UpdateInstanceArtifactsOptions, body: UpdateInstanceArtifactsPayload) {
+    debug('PUT %s %O', recordRoutes.instanceArtifacts(options.instanceId), body)
```
Comment (ryanthemanuel) [inline packages/server/lib/cloud/api/index.ts]: I think we inadvertently put capital O in a couple of places in the protocol code, but I think the pattern I've seen elsewhere is the little o. It makes objects easier to copy and paste later.

```suggestion
    debug('PUT %s %o', recordRoutes.instanceArtifacts(options.instanceId), body)
```

File: packages/server/lib/cloud/protocol.ts [hunk 3]
```diff
@@ -170,7 +174,31 @@ export class ProtocolManager implements ProtocolManagerShape {
   }
 
   async afterSpec () {
-    await this.invokeAsync('afterSpec', { isEssential: true })
+    const startTime = performance.now() + performance.timeOrigin
+
+    debug({ startTime })
```
Comment (ryanthemanuel) [inline packages/server/lib/cloud/protocol.ts]: Should we have a description here around what this is?

File: packages/server/lib/cloud/protocol.ts [hunk 4]
```diff
@@ -277,11 +305,7 @@ export class ProtocolManager implements ProtocolManagerShape {
     }
   }
 
-  async uploadCaptureArtifact ({ uploadUrl, fileSize, filePath }: CaptureArtifact): Promise<{
-    success: boolean
-    fileSize: number | bigint
-    specAccess: ReturnType<AppCaptureProtocolInterface['getDbMetadata']>
-  } | undefined> {
+  async uploadCaptureArtifact ({ uploadUrl, fileSize, filePath }: CaptureArtifact, captureErrorsOverride?: boolean) {
```
Comment (ryanthemanuel) [inline packages/server/lib/cloud/protocol.ts]: Is captureErrorsOverride just used for testing purposes?

File: packages/server/lib/cloud/protocol.ts (hunk 3 above)
Comment (cacieprins) [reply packages/server/lib/cloud/protocol.ts]: I think this debug can be removed- the important debug is on L191

Comment (cacieprins) [inline packages/server/lib/cloud/protocol.ts]:
```suggestion
```

File: packages/server/lib/cloud/protocol.ts (hunk 4 above)
Comment (cacieprins) [reply packages/server/lib/cloud/protocol.ts]: Yep. Could refactor a bit to have caller pass in the env var to make things a little cleaner.

**Review by mschile — APPROVED** [review]

**Review by ryanthemanuel — APPROVED** [review]

**Pull request merged**